astropy/cosmology/units.py

```diff
@@ -77,15 +93,24 @@ def redshift_temperature(cosmology=None, **atzkw):
     with default_cosmology.set(cosmology):  # if already cosmo, passes through
         cosmology = default_cosmology.get()
 
-    def z_to_Tcmb(z):
-        return cosmology.Tcmb(z)
+    allowed_distances = ('comoving', 'lookback', 'luminosity')
+    if distance_kind not in allowed_distances:
+        raise ValueError(f"`distance_kind` is not one of {allowed_distances}")
 
-    def Tcmb_to_z(T):
-        return z_at_value(cosmology.Tcmb, T << u.K, **atzkw)
+    method_name = distance_kind + "_distance"
 
-    return u.Equivalency([(redshift, u.K, z_to_Tcmb, Tcmb_to_z)],
-                         "redshift_temperature",
-                         {'cosmology': cosmology})
+    def z_to_distance(z):
+        """Redshift to distance."""
+        return getattr(cosmology, method_name)(z)
+
+    def distance_to_z(d):
+        """Distance to redshift."""
+        method = getattr(cosmology, method_name)
```

Comment:
Maybe take method out of this function and use it in `z_to_distance` as well?

Reply:
That's a nice simplification. Thanks.